x-pack/legacy/plugins/xpack_main/public/services/xpack_info.js

```diff
@@ -4,48 +4,48 @@
  */
 
-import { get } from 'lodash';
+import { get as lodashGet } from 'lodash';
 import chrome from 'ui/chrome';
 import { xpackInfoSignature } from './xpack_info_signature';
 import { convertKeysToCamelCaseDeep } from '../../../../server/lib/key_case_converter';
 
 const XPACK_INFO_KEY = 'xpackMain.info';
 
-export function XPackInfoProvider($window, $injector) {
+let inProgressRefreshPromise = null;
 
-  class XPackInfo {
-    constructor(initialInfo = {}) {
-      this.inProgressRefreshPromise = null;
-      this.setAll(initialInfo);
-    }
+const setAll = (updatedXPackInfo) => {
+  // The decision to convert kebab-case/snake-case keys to camel-case keys stemmed from an old
+  // convention of using kebabe-case/snake-case in API response bodies but camel-case in JS
+  // objects. See pull #29304 for more info.
+  const camelCasedXPackInfo = convertKeysToCamelCaseDeep(updatedXPackInfo);
+  sessionStorage.setItem(XPACK_INFO_KEY, JSON.stringify(camelCasedXPackInfo));
+};
 
-    get = (path, defaultValue = undefined) => {
-      const xpackInfoValuesJson = $window.sessionStorage.getItem(XPACK_INFO_KEY);
-      const xpackInfoValues = xpackInfoValuesJson ? JSON.parse(xpackInfoValuesJson) : {};
-      return get(xpackInfoValues, path, defaultValue);
-    };
+const get = (path, defaultValue = undefined) => {
+  const xpackInfoValuesJson = sessionStorage.getItem(XPACK_INFO_KEY);
+  const xpackInfoValues = xpackInfoValuesJson ? JSON.parse(xpackInfoValuesJson) : {};
+  return lodashGet(xpackInfoValues, path, defaultValue);
+};
 
-    setAll = (updatedXPackInfo) => {
-      // The decision to convert kebab-case/snake-case keys to camel-case keys stemmed from an old
-      // convention of using kebabe-case/snake-case in API response bodies but camel-case in JS
-      // objects. See pull #29304 for more info.
-      const camelCasedXPackInfo = convertKeysToCamelCaseDeep(updatedXPackInfo);
-      $window.sessionStorage.setItem(XPACK_INFO_KEY, JSON.stringify(camelCasedXPackInfo));
-    };
+export function xpackInfoService($injector) {
+  setAll(chrome.getInjected('xpackInitialInfo') || {});
+  return {
+    get,
+    setAll,
 
-    clear = () => {
-      $window.sessionStorage.removeItem(XPACK_INFO_KEY);
-    };
+    clear: () => {
+      sessionStorage.removeItem(XPACK_INFO_KEY);
+    },
 
-    refresh = () => {
-      if (this.inProgressRefreshPromise) {
-        return this.inProgressRefreshPromise;
+    refresh: () => {
+      if (inProgressRefreshPromise) {
+        return inProgressRefreshPromise;
       }
 
       // store the promise in a shared location so that calls to
       // refresh() before this is complete will get the same promise
-      const $http = $injector.get('$http');
-      this.inProgressRefreshPromise = (
+      const $http = $injector.get('$http'); // $http must be consumed here - if passed in, circular dependency errors
+      inProgressRefreshPromise = (
         $http.get(chrome.addBasePath('/api/xpack/v1/info'))
```

Comment:
use kfetch ?

Reply:
and then you don't need to pass $injector in

Reply:
Maybe try `core.http`

```js
import { npStart } from 'ui/new_platform';

npStart.core.http
```